### PR TITLE
Improve docs on contourf extend

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Based on #13038 I've rewritten the docstring for `contourf` `extend` to better explain how to use it.

### CODE REVIEW

**Comment by jklymak** [comment]
This seems fine, but isn’t the underlying issue that the contour is not being updated automatically?

**Comment by timhoffm** [comment]
> This seems fine, but isn’t the underlying issue that the contour is not being updated automatically?


I don‘t know if that‘s possible. Does the cmap know about the associated scalar mappable? Otherwise there‘s no way that it can notify the QuadContourSet that it needs to update.

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -1766,14 +1766,41 @@ def _initialize_x_y(self, z):
...
+
+            Example::
+
+                x = np.arange(1, 10)
+                y = x.reshape(-1, 1)
+                h = x * y
+
+                cs = plt.contourf(h, levels=[10, 30, 50],
+                    colors=['#808080', '#A0A0A0', '#C0C0C0'], extend='both')
+                cs.cmap.set_over('red')
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply lib/matplotlib/contour.py]: Updating a `QuadContourSet` on each draw would essentially mean that we have to run the full code of `changed()`. Not quite sure if we want to do this.

[the remark this answers is not in the document]
Comment (tacaswell) [reply lib/matplotlib/contour.py]: I agree this is the conservative path here.

**Comment by jklymak** [comment]
> I don‘t know if that‘s possible. Does the cmap know about the associated scalar mappable? Otherwise there‘s no way that it can notify the QuadContourSet that it needs to update.

I guess not.  I wonder if what is actually needed is a `cs.set_over`, `cs.set_under` method if users really need to do this after the fact?

**Comment by timhoffm** [comment]
I guess not. cs.set_under and similar seem counter-intuitive, when there‘s already a cs.cmap.set_under with a different behavior. It‘s not obvious that one can use one but not the other. That would actually be my least favorite solution if we‘re going to change the API.

Other possible API changes: It‘s actually not reasonable to define the over/under colors at another time compared to the regular colors. Therefore one could:

- add additional kwargs for over/under to contourf, or
- Make it reasonably simple to create a cmap and norm for a discrete set of colors and boundaries. These could then be supplied to contourf.

**Comment by timhoffm** [comment]
Note: Travis failure is due to #13137 and unrelated to this PR.

**Pull request merged**